A spec with an empty `content` block will always fail to validate

In the case that an OpenAPI spec is not defining the response body - which can be a valid thing to do - we receive an error:

```
validator.go:50: Error: GET / 200 operation response content type '' does not exist, Reason: The content type '' of the GET response received has not been defined, it's an unknown type, Line: 11, Column: 11
```

For spec:

```yaml
openapi: "3.0.0"
info:
  title: Healthcheck
  version: '0.1.0'
paths:
  /health:
    get:
      responses:
        '200':
          description: pet response
          content: {}
```

There's quite a few other folks doing this, too https://github.com/search?q=%22content%3A+%7B%7D%22+language%3Ayaml&type=code so not sure if maybe this needs to be something we can opt-in to allowing maybe? Not sure if it's something we want to have on-by-default

--- a/responses/validate_body.go
+++ b/responses/validate_body.go
@@ -4,151 +4,151 @@
 package responses
 
 import (
-    "github.com/pb33f/libopenapi-validator/errors"
-    "github.com/pb33f/libopenapi-validator/helpers"
-    "github.com/pb33f/libopenapi-validator/paths"
-    "github.com/pb33f/libopenapi/datamodel/high/base"
-    "github.com/pb33f/libopenapi/datamodel/high/v3"
-    "github.com/pb33f/libopenapi/utils"
-    "net/http"
-    "strconv"
-    "strings"
+	"github.com/pb33f/libopenapi-validator/errors"
+	"github.com/pb33f/libopenapi-validator/helpers"
+	"github.com/pb33f/libopenapi-validator/paths"
+	"github.com/pb33f/libopenapi/datamodel/high/base"
+	"github.com/pb33f/libopenapi/datamodel/high/v3"
+	"github.com/pb33f/libopenapi/utils"
+	"net/http"
+	"strconv"
+	"strings"
 )
 
 func (v *responseBodyValidator) ValidateResponseBody(
-    request *http.Request,
-    response *http.Response) (bool, []*errors.ValidationError) {
+	request *http.Request,
+	response *http.Response) (bool, []*errors.ValidationError) {
 
-    // find path
-    var pathItem *v3.PathItem
-    var errs []*errors.ValidationError
-    if v.pathItem == nil {
-        pathItem, errs, _ = paths.FindPath(request, v.document)
-        if pathItem == nil || errs != nil {
-            v.errors = errs
-            return false, errs
-        }
-    } else {
-        pathItem = v.pathItem
-    }
+	// find path
+	var pathItem *v3.PathItem
+	var errs []*errors.ValidationError
+	if v.pathItem == nil {
+		pathItem, errs, _ = paths.FindPath(request, v.document)
+		if pathItem == nil || errs != nil {
+			v.errors = errs
+			return false, errs
+		}
+	} else {
+		pathItem = v.pathItem
+	}
 
-    var validationErrors []*errors.ValidationError
-    operation := helpers.ExtractOperation(request, pathItem)
+	var validationErrors []*errors.ValidationError
+	operation := helpers.ExtractOperation(request, pathItem)
 
-    // extract the response code from the response
-    httpCode := response.StatusCode
-    contentType := response.Header.Get(helpers.ContentTypeHeader)
+	// extract the response code from the response
+	httpCode := response.StatusCode
+	contentType := response.Header.Get(helpers.ContentTypeHeader)
 
-    // extract the media type from the content type header.
-    mediaTypeSting, _, _ := helpers.ExtractContentType(contentType)
+	// extract the media type from the content type header.
+	mediaTypeSting, _, _ := helpers.ExtractContentType(contentType)
 
-    // check if the response code is in the contract
-    foundResponse := operation.Responses.FindResponseByCode(httpCode)
-    if foundResponse != nil {
+	// check if the response code is in the contract
+	foundResponse := operation.Responses.FindResponseByCode(httpCode)
+	if foundResponse != nil {
 
-        // check content type has been defined in the contract
-        if mediaType, ok := foundResponse.Content[mediaTypeSting]; ok {
+		// check content type has been defined in the contract
+		if mediaType, ok := foundResponse.Content[mediaTypeSting]; ok {
 
-            validationErrors = append(validationErrors,
-                v.checkResponseSchema(request, response, mediaTypeSting, mediaType)...)
+			validationErrors = append(validationErrors,
+				v.checkResponseSchema(request, response, mediaTypeSting, mediaType)...)
 
-        } else {
+		} else {
 
-            // check that the operation *actually* returns a body. (i.e. a 204 response)
-            if foundResponse.Content != nil {
+			// check that the operation *actually* returns a body. (i.e. a 204 response)
+			if foundResponse.Content != nil && len(foundResponse.Content) > 0 {
 
-                // content type not found in the contract
-                codeStr := strconv.Itoa(httpCode)
-                validationErrors = append(validationErrors,
-                    errors.ResponseContentTypeNotFound(operation, request, response, codeStr, false))
+				// content type not found in the contract
+				codeStr := strconv.Itoa(httpCode)
+				validationErrors = append(validationErrors,
+					errors.ResponseContentTypeNotFound(operation, request, response, codeStr, false))
 
-            }
-        }
-    } else {
+			}
+		}
+	} else {
 
-        // no code match, check for default response
-        if operation.Responses.Default != nil {
+		// no code match, check for default response
+		if operation.Responses.Default != nil {
 
-            // check content type has been defined in the contract
-            if mediaType, ok := operation.Responses.Default.Content[mediaTypeSting]; ok {
+			// check content type has been defined in the contract
+			if mediaType, ok := operation.Responses.Default.Content[mediaTypeSting]; ok {
 
-                validationErrors = append(validationErrors,
-                    v.checkResponseSchema(request, response, contentType, mediaType)...)
+				validationErrors = append(validationErrors,
+					v.checkResponseSchema(request, response, contentType, mediaType)...)
 
-            } else {
+			} else {
 
-                // check that the operation *actually* returns a body. (i.e. a 204 response)
-                if operation.Responses.Default.Content != nil {
+				// check that the operation *actually* returns a body. (i.e. a 204 response)
+				if operation.Responses.Default.Content != nil && len(operation.Responses.Default.Content) > 0 {
 
-                    // content type not found in the contract
-                    codeStr := strconv.Itoa(httpCode)
-                    validationErrors = append(validationErrors,
-                        errors.ResponseContentTypeNotFound(operation, request, response, codeStr, true))
-                }
-            }
+					// content type not found in the contract
+					codeStr := strconv.Itoa(httpCode)
+					validationErrors = append(validationErrors,
+						errors.ResponseContentTypeNotFound(operation, request, response, codeStr, true))
+				}
+			}
 
-        } else {
-            // TODO: add support for '2XX' and '3XX' responses in the contract
-            // no default, no code match, nothing!
-            validationErrors = append(validationErrors,
-                errors.ResponseCodeNotFound(operation, request, httpCode))
-        }
-    }
-    if len(validationErrors) > 0 {
-        return false, validationErrors
-    }
-    return true, nil
+		} else {
+			// TODO: add support for '2XX' and '3XX' responses in the contract
+			// no default, no code match, nothing!
+			validationErrors = append(validationErrors,
+				errors.ResponseCodeNotFound(operation, request, httpCode))
+		}
+	}
+	if len(validationErrors) > 0 {
+		return false, validationErrors
+	}
+	return true, nil
 }
 
 func (v *responseBodyValidator) checkResponseSchema(
-    request *http.Request,
-    response *http.Response,
-    contentType string,
-    mediaType *v3.MediaType) []*errors.ValidationError {
+	request *http.Request,
+	response *http.Response,
+	contentType string,
+	mediaType *v3.MediaType) []*errors.ValidationError {
 
-    var validationErrors []*errors.ValidationError
+	var validationErrors []*errors.ValidationError
 
-    // currently, we can only validate JSON based responses, so check for the presence
-    // of 'json' in the content type (what ever it may be) so we can perform a schema check on it.
-    // anything other than JSON, will be ignored.
-    if strings.Contains(strings.ToLower(contentType), helpers.JSONType) {
+	// currently, we can only validate JSON based responses, so check for the presence
+	// of 'json' in the content type (what ever it may be) so we can perform a schema check on it.
+	// anything other than JSON, will be ignored.
+	if strings.Contains(strings.ToLower(contentType), helpers.JSONType) {
 
-        // extract schema from media type
-        if mediaType.Schema != nil {
+		// extract schema from media type
+		if mediaType.Schema != nil {
 
-            var schema *base.Schema
-            var renderedInline, renderedJSON []byte
+			var schema *base.Schema
+			var renderedInline, renderedJSON []byte
 
-            // have we seen this schema before? let's hash it and check the cache.
-            hash := mediaType.GoLow().Schema.Value.Hash()
+			// have we seen this schema before? let's hash it and check the cache.
+			hash := mediaType.GoLow().Schema.Value.Hash()
 
-            if cacheHit, ch := v.schemaCache[hash]; ch {
+			if cacheHit, ch := v.schemaCache[hash]; ch {
 
-                // got a hit, use cached values
-                schema = cacheHit.schema
-                renderedInline = cacheHit.renderedInline
-                renderedJSON = cacheHit.renderedJSON
+				// got a hit, use cached values
+				schema = cacheHit.schema
+				renderedInline = cacheHit.renderedInline
+				renderedJSON = cacheHit.renderedJSON
 
-            } else {
+			} else {
 
-                // render the schema inline and perform the intensive work of rendering and converting
-                // this is only performed once per schema and cached in the validator.
-                schema = mediaType.Schema.Schema()
-                renderedInline, _ = schema.RenderInline()
-                renderedJSON, _ = utils.ConvertYAMLtoJSON(renderedInline)
-                v.schemaCache[hash] = &schemaCache{
-                    schema:         schema,
-                    renderedInline: renderedInline,
-                    renderedJSON:   renderedJSON,
-                }
-            }
+				// render the schema inline and perform the intensive work of rendering and converting
+				// this is only performed once per schema and cached in the validator.
+				schema = mediaType.Schema.Schema()
+				renderedInline, _ = schema.RenderInline()
+				renderedJSON, _ = utils.ConvertYAMLtoJSON(renderedInline)
+				v.schemaCache[hash] = &schemaCache{
+					schema:         schema,
+					renderedInline: renderedInline,
+					renderedJSON:   renderedJSON,
+				}
+			}
 
-            // render the schema, to be used for validation
-            valid, vErrs := ValidateResponseSchema(request, response, schema, renderedInline, renderedJSON)
-            if !valid {
-                validationErrors = append(validationErrors, vErrs...)
-            }
-        }
-    }
-    return validationErrors
+			// render the schema, to be used for validation
+			valid, vErrs := ValidateResponseSchema(request, response, schema, renderedInline, renderedJSON)
+			if !valid {
+				validationErrors = append(validationErrors, vErrs...)
+			}
+		}
+	}
+	return validationErrors
 }

--- a/responses/validate_body_test.go
+++ b/responses/validate_body_test.go
@@ -945,3 +945,46 @@ paths:
 	assert.Len(t, errors[0].SchemaValidationErrors, 2)
 	assert.Equal(t, "200 response body for '/burgers/createBurger' failed to validate schema", errors[0].Message)
 }
+
+func TestValidateBody_EmptyContentType_Valid(t *testing.T) {
+	spec := `openapi: "3.0.0"
+info:
+  title: Healthcheck
+  version: '0.1.0'
+paths:
+  /health:
+    get:
+      responses:
+        '200':
+          description: pet response
+          content: {}`
+
+	doc, _ := libopenapi.NewDocument([]byte(spec))
+
+	m, _ := doc.BuildV3Model()
+	v := NewResponseBodyValidator(&m.Model)
+
+	// build a request
+	request, _ := http.NewRequest(http.MethodGet, "https://things.com/health", nil)
+
+	// simulate a request/response
+	res := httptest.NewRecorder()
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set(helpers.ContentTypeHeader, helpers.JSONContentType)
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write(nil)
+	}
+
+	// fire the request
+	handler(res, request)
+
+	// record response
+	response := res.Result()
+
+	// validate!
+	valid, errors := v.ValidateResponseBody(request, response)
+
+	assert.True(t, valid)
+	assert.Len(t, errors, 0)
+
+}

